src/views/tickets/Ticket.js

```diff
@@ -12,14 +15,17 @@ export default class Ticket extends Component {
         ticketID: props.match.params.ticketID,
         processed: false,
       },
-      refreshInterval: setInterval(this.refreshTicketState, REFRESH_EVERY_MS),
+      currentCode: '',
```

Comment:
Please move refreshInterval inline again so it's clear that it's part of state - internally react uses some special hooks on setting the state variables, more efficient and clear to have it all in one place.

Reply:
I don't see the reason why `refreshInterval` should be a state. Variables in state should be regarded as a state corresponding to the rendering. When a variable in state changes, the component will re-render itself. We use `state.ticketInfo.processed` to trigger the re-rendering.

Reply:
That’s a good point! Having it as a class-scoped variable makes more sense, then.